src/main/cluster-detectors/base-cluster-detector.ts

```diff
@@ -15,14 +15,14 @@ export interface ClusterDetectionResult {
 export class BaseClusterDetector {
   key: string;
 
-  constructor(public cluster: Cluster) {
+  constructor(public cluster: Cluster, private _k8sRequest: K8sRequest) {
```

Comment:
Why not just have this as a dependency? We haven't exported these at all anywhere so we can break the constructor all we like.